### PR TITLE
Updating the model on app changes, e.g reloads

### LINKED ISSUE #40: React on a suspend notification
When triggering a reload in the same engine session a suspend notification is triggered in catwalk but a resume call is never issued.

### PR DESCRIPTION
**Pull request by hrigner**
closes #40

### CODE REVIEW

File: src/components/model.jsx [hunk 1]
```diff
@@ -27,19 +27,16 @@ export class Model extends React.Component {
   }
 
   async componentDidMount() {
+    this.updatePlay();
```
Comment (peol) [inline src/components/model.jsx]: Curious, why do we call it "play"?

File: src/components/model.jsx [hunk 2]
```diff
@@ -249,4 +265,4 @@ Model.defaultProps = {
   model: null,
 };
 
-export default withApp(withModel(Model, async app => app.getTablesAndKeys({}, {}, 0, true, false)));
+export default withApp(withModel(Model, async app => app.getTablesAndKeys({}, {}, 0, true, false), true));
```
Comment (peol) [inline src/components/model.jsx]: Can we use an object into `withModel` instead so it's clearer what the parameters do?

File: src/components/with-model.jsx [hunk 3]
```diff
@@ -21,10 +21,13 @@ function withModel(WrappedComponent, createModel) {
       this.setState = () => {};
     }
 
-    async updateModel() {
+    async updateModel(firstTime) {
       const { app } = this.props;
       try {
         const model = await createModel(app, this.props);
+        if (firstTime && updateOnAppInvalidation) {
+          app.on('changed', this.updateModel);
```
Comment (peol) [inline src/components/with-model.jsx]: We need to remove the listener on unmount/destroy too (`app.removeListener('changed', fn)`)

File: src/components/model.jsx (hunk 2 above)
Comment (hrigner) [reply src/components/model.jsx]: Yes, good point!

File: src/components/with-model.jsx (hunk 3 above)
Comment (hrigner) [reply src/components/with-model.jsx]: thanks!

**Comment by mindspank** [comment]
woho!

**Pull request merged**